lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -183,7 +183,11 @@ module.exports = {
     const resource = this.serverless.service.provider.compiledCloudFormationTemplate.Resources[
       this.provider.naming.getRoleLogicalId()
     ].Properties;
-    if (!_.has(resource, 'ManagedPolicyArns') || _.isEmpty(resource.ManagedPolicyArns)) {
+
+    if (
+      !_.has(resource, 'ManagedPolicyArns') ||
+      typeof resource.ManagedPolicyArns === 'undefined'
```

Comment:
Let's replace this condition simply with `if (!Array.isArray(resource.ManagedPolicyArns))`